processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -385,49 +402,19 @@ public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec)
     @Override
     public FloatColumnSelector makeFloatColumnSelector(String columnName)
     {
-      FloatColumnSelector existing = floatColumnSelectorMap.get(columnName);
-      if (existing != null) {
-        return existing;
-      } else {
-        FloatColumnSelector newSelector = delegate.makeFloatColumnSelector(columnName);
-        FloatColumnSelector prev = floatColumnSelectorMap.putIfAbsent(
-            columnName,
-            newSelector
-        );
-        return prev != null ? prev : newSelector;
-      }
+      return floatColumnSelectorMap.computeIfAbsent(columnName, delegate::makeFloatColumnSelector);
```

Comment:
Because of how code is inlined it might be not identical. And for ConcurrentHashMap the difference is bigger, because computeIfAbsent() always locks, and get() just makes a volatile read.